2024-05-03 16:18:46,386 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 16:19:21,417 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/urls.py changed, reloading.
2024-05-03 16:19:21,585 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 16:19:28,840 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/urls.py changed, reloading.
2024-05-03 16:19:29,006 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 16:19:48,572 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/urls.py changed, reloading.
2024-05-03 16:19:48,730 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 16:20:36,214 (WARNING) - django.request - Unauthorized: /api/products/
2024-05-03 16:20:36,218 (WARNING) - django.request - Forbidden: /api/products/
2024-05-03 16:20:36,220 (WARNING) - django.request - Bad Request: /api/products/
2024-05-03 16:20:36,231 (WARNING) - django.request - Not Found: /api/products/fbf290e4-3078-4240-9f29-e84e0c38e0ab/
2024-05-03 16:21:24,025 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings/common.py changed, reloading.
2024-05-03 16:21:24,219 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 16:21:27,298 (WARNING) - django.request - Unauthorized: /api/products/
2024-05-03 16:21:27,306 (WARNING) - django.request - Forbidden: /api/products/
2024-05-03 16:21:27,313 (WARNING) - django.request - Bad Request: /api/products/
2024-05-03 16:21:27,340 (WARNING) - django.request - Not Found: /api/products/3a372917-dd5e-45fb-90d2-ce05f9dfcf2f/
2024-05-03 16:28:02,392 (WARNING) - django.request - Not Found: /auth/jwt/token/
2024-05-03 16:28:22,632 (WARNING) - django.request - Method Not Allowed: /auth/jwt/create/
2024-05-03 16:28:35,330 (WARNING) - django.request - Unauthorized: /auth/jwt/create/
2024-05-03 16:28:49,752 (WARNING) - django.request - Unauthorized: /auth/jwt/create/
2024-05-03 16:35:10,483 (WARNING) - django.request - Not Found: /api/products/fd54abe1-f8a1-4c3e-befd-ecb7af6f0406/image
2024-05-03 17:40:06,766 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 18:24:57,505 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 18:48:12,895 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 18:52:57,740 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 18:53:00,235 (WARNING) - django.request - Not Found: /favicon.ico
2024-05-03 19:01:09,293 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:01:09,471 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:01:10,660 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:01:10,859 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:01:13,055 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:01:13,236 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:02:52,493 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:02:52,689 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:02,058 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:03:02,265 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:05,505 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:03:05,664 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:07,858 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:03:08,025 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:10,218 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:03:10,390 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:43,314 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:03:43,515 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:03:46,723 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:07:59,677 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:08:38,661 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:09:10,632 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:10:47,035 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:11:52,210 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:11:52,394 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:12:00,762 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:12:00,936 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:12:02,098 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:12:02,265 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:12:05,465 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:12:05,638 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:12:06,814 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:12:06,982 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:12:10,212 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:12:10,386 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:13:15,509 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:13:15,781 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:13:28,244 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:13:28,431 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:13:29,607 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:13:29,796 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:21:06,555 (WARNING) - django.request - Not Found: /static/core/styles.css/
2024-05-03 19:21:06,559 (WARNING) - django.request - Not Found: /static/core/logo.png/
2024-05-03 19:21:08,728 (WARNING) - django.request - Not Found: /static/core/styles.css/
2024-05-03 19:21:08,735 (WARNING) - django.request - Not Found: /static/core/logo.png/
2024-05-03 19:22:32,104 (WARNING) - django.request - Not Found: /static/core/styles.css/
2024-05-03 19:22:32,107 (WARNING) - django.request - Not Found: /static/core/logo.png/
2024-05-03 19:22:44,591 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:26:11,255 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-03 19:26:11,541 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:30:11,238 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 19:48:27,393 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-03 20:06:24,938 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:06:24,944 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:06:24,946 (WARNING) - django.request - Not Found: /api/orderitems/
2024-05-03 20:06:24,954 (WARNING) - django.request - Not Found: /api/orderitems/
2024-05-03 20:13:58,591 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:06,655 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:06,657 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:06,658 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:06,659 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:23,068 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:23,069 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:23,071 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:14:23,073 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-03 20:21:06,747 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-05 23:56:11,573 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-05 23:56:13,745 (WARNING) - django.request - Not Found: /favicon.ico
2024-05-06 19:12:56,880 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-06 19:16:56,391 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-06 19:16:59,620 (WARNING) - django.request - Not Found: /favicon.ico
2024-05-06 19:18:54,323 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-06 19:18:54,328 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-06 19:18:54,331 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-06 19:18:54,339 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-06 19:40:40,189 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-06 19:47:28,320 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-06 19:47:28,332 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-06 19:50:09,958 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-06 19:52:39,605 (ERROR) - django.request - Internal Server Error: /api/carts/[object Promise]/items/fd54abe1-f8a1-4c3e-befd-ecb7af6f0406
Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/utils/deprecation.py", line 136, in __call__
    response = self.process_response(request, response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/middleware/common.py", line 108, in process_response
    return self.response_redirect_class(self.get_full_path_with_slash(request))
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/middleware/common.py", line 87, in get_full_path_with_slash
    raise RuntimeError(
RuntimeError: You called this URL via POST, but the URL doesn't end in a slash and you have APPEND_SLASH set. Django can't redirect to the slash URL while maintaining POST data. Change your form to point to 127.0.0.1:8000/api/carts/%5Bobject%20Promise%5D/items/fd54abe1-f8a1-4c3e-befd-ecb7af6f0406/ (note the trailing slash), or set APPEND_SLASH=False in your Django settings.
2024-05-06 20:04:14,520 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 14:11:50,551 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 14:14:28,125 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-07 14:14:28,133 (WARNING) - django.request - Not Found: /api/cart/items/
2024-05-07 16:12:57,568 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:14:31,334 (WARNING) - django.request - Not Found: /favicon.ico
2024-05-07 16:14:39,049 (WARNING) - django.request - Not Found: /ap1/
2024-05-07 16:14:43,548 (WARNING) - django.request - Not Found: /ap1/
2024-05-07 16:16:23,375 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:16:23,620 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:16:24,843 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:16:24,998 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:16:37,471 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:16:37,635 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:17:13,085 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:17:49,686 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:24:41,381 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:37:33,554 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:37:57,410 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:37:57,588 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-07 16:38:03,849 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-07 16:38:04,008 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 13:14:09,342 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 13:16:10,114 (WARNING) - django.request - Unauthorized: /api/customers/
2024-05-08 13:16:33,434 (WARNING) - django.request - Unauthorized: /auth/users/
2024-05-08 13:16:55,926 (WARNING) - django.request - Not Found: /auth/jwt/
2024-05-08 13:17:14,767 (WARNING) - django.request - Method Not Allowed: /auth/jwt/create
2024-05-08 13:17:24,664 (WARNING) - django.request - Unauthorized: /auth/jwt/create
2024-05-08 13:17:25,969 (WARNING) - django.request - Unauthorized: /auth/jwt/create
2024-05-08 13:18:37,904 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 13:18:51,832 (WARNING) - django.request - Method Not Allowed: /auth/jwt/create
2024-05-08 13:20:11,451 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 13:26:59,923 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 13:27:14,953 (ERROR) - django.request - Internal Server Error: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/
Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/viewsets.py", line 124, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 56, in retrieve
    return Response(serializer.data)
                    ^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 571, in data
    ret = super().data
          ^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 249, in data
    self._data = self.to_representation(self.instance)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 538, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 713, in to_representation
    return [
           ^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 714, in <listcomp>
    self.child.to_representation(item) for item in iterable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 538, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 523, in to_representation
    for field in fields:
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 384, in _readable_fields
    for field in self.fields.values():
                 ^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 372, in fields
    for key, value in self.get_fields().items():
                      ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1103, in get_fields
    field_class, field_kwargs = self.build_field(
                                ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1249, in build_field
    return self.build_unknown_field(field_name, model_class)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1367, in build_unknown_field
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: Field name `image` is not valid for model `Product` in `api.serializers.CartItemProductSerializer`.
2024-05-08 13:27:32,342 (WARNING) - django.request - Not Found: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/cartitems
2024-05-08 13:27:44,208 (ERROR) - django.request - Internal Server Error: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/items/
Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/viewsets.py", line 124, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 43, in list
    return self.get_paginated_response(serializer.data)
                                       ^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 795, in data
    ret = super().data
          ^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 249, in data
    self._data = self.to_representation(self.instance)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 713, in to_representation
    return [
           ^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 714, in <listcomp>
    self.child.to_representation(item) for item in iterable
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 538, in to_representation
    ret[field.field_name] = field.to_representation(attribute)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 523, in to_representation
    for field in fields:
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 384, in _readable_fields
    for field in self.fields.values():
                 ^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/utils/functional.py", line 47, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 372, in fields
    for key, value in self.get_fields().items():
                      ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1103, in get_fields
    field_class, field_kwargs = self.build_field(
                                ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1249, in build_field
    return self.build_unknown_field(field_name, model_class)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/serializers.py", line 1367, in build_unknown_field
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: Field name `image` is not valid for model `Product` in `api.serializers.CartItemProductSerializer`.
2024-05-08 13:28:30,111 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 13:28:30,379 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 13:30:32,769 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 13:30:32,947 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 13:30:35,088 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 13:30:35,247 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 14:30:30,847 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/admin.py changed, reloading.
2024-05-08 14:30:31,190 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 14:31:06,075 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 15:14:22,285 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 17:36:08,260 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 17:36:22,597 (WARNING) - django.request - Unauthorized: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/items/
2024-05-08 17:36:36,233 (WARNING) - django.request - Method Not Allowed: /auth/jwt/refresh
2024-05-08 17:42:43,336 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:43:05,194 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:46:50,159 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:47:24,548 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:47:58,081 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:48:51,615 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:49:30,893 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:50:14,075 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:55:38,913 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 17:56:01,348 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 18:00:24,430 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 18:00:47,537 (WARNING) - django.request - Bad Request: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 18:37:00,150 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:37:17,230 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:37:37,586 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:37:53,801 (WARNING) - django.request - Unauthorized: /api/products/
2024-05-08 18:37:53,805 (WARNING) - django.request - Unauthorized: /api/products/
2024-05-08 18:37:56,545 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:37:56,549 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:38:29,796 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 18:38:36,445 (WARNING) - django.request - Unauthorized: /api/
2024-05-08 18:45:44,029 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:45:44,548 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:45:58,023 (WARNING) - django.request - Unauthorized: /api/carts/
2024-05-08 18:46:08,186 (WARNING) - django.request - Method Not Allowed: /auth/jwt/refresh
2024-05-08 18:46:56,223 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 18:58:10,286 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:10,531 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:14,813 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:15,025 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:26,420 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:26,585 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:42,126 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:42,301 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:43,404 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:43,561 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:44,663 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:44,839 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:45,937 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:46,141 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:49,289 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:49,451 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:52,589 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:52,772 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:58:53,882 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:58:54,075 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:04,507 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:04,710 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:06,944 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:07,172 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:12,517 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:12,704 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:44,765 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:44,933 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:47,169 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:47,338 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 18:59:57,760 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 18:59:57,930 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 19:00:02,108 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 19:00:02,277 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 19:00:05,524 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 19:00:05,701 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 19:00:06,859 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py changed, reloading.
2024-05-08 19:00:07,026 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 19:15:12,568 (WARNING) - django.request - Method Not Allowed: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
2024-05-08 19:15:54,141 (WARNING) - django.request - Not Found: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/1
2024-05-08 19:23:08,484 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/[object Object]/
2024-05-08 19:46:42,895 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 19:46:42,898 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 19:46:47,200 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 19:46:47,202 (WARNING) - django.request - Unauthorized: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 19:47:05,139 (WARNING) - django.request - Unauthorized: /auth/jwt/create/
2024-05-08 19:52:02,792 (WARNING) - django.request - Unauthorized: /api/
2024-05-08 19:52:14,270 (WARNING) - django.request - Method Not Allowed: /auth/jwt/refresh
2024-05-08 19:52:46,143 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-08 19:52:46,411 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 19:52:48,659 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-08 19:52:48,853 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 20:01:17,650 (ERROR) - django.dispatch - Error calling on_order_created in Signal.send_robust() (print() argument after ** must be a mapping, not Order)
Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/dispatch/dispatcher.py", line 306, in send_robust
    response = receiver(signal=self, sender=sender, **named)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/core/signals/handlers.py", line 7, in on_order_created
    print(**kwargs['order'])
TypeError: print() argument after ** must be a mapping, not Order
2024-05-08 20:21:39,636 (WARNING) - django.request - Not Found: /api/orders/{}/
2024-05-08 20:22:32,708 (WARNING) - django.request - Not Found: /api/orders/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:22:34,586 (WARNING) - django.request - Not Found: /api/orders/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:23:50,233 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:23:52,113 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:24:06,959 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:24:08,838 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:24:25,902 (ERROR) - django.request - Internal Server Error: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/items/
Traceback (most recent call last):
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py", line 90, in save
    cart_item = CartItem.objects.get(
                ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 649, in get
    raise self.model.DoesNotExist(
api.models.CartItem.DoesNotExist: CartItem matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.ForeignKeyViolation: insert or update on table "api_cartitem" violates foreign key constraint "api_cartitem_cart_id_26c2013b_fk_api_cart_id"
DETAIL:  Key (cart_id)=(31e197b4-dd56-402a-bd35-8e4f984a7d92) is not present in table "api_cart".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/viewsets.py", line 124, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 19, in create
    self.perform_create(serializer)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 24, in perform_create
    serializer.save()
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py", line 99, in save
    self.instance = CartItem.objects.create(
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: insert or update on table "api_cartitem" violates foreign key constraint "api_cartitem_cart_id_26c2013b_fk_api_cart_id"
DETAIL:  Key (cart_id)=(31e197b4-dd56-402a-bd35-8e4f984a7d92) is not present in table "api_cart".

2024-05-08 20:24:30,648 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:24:32,522 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:25:07,252 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 20:26:56,466 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:26:58,528 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:02,531 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:03,370 (ERROR) - django.dispatch - Error calling on_order_created in Signal.send_robust() (print() argument after ** must be a mapping, not Order)
Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/dispatch/dispatcher.py", line 306, in send_robust
    response = receiver(signal=self, sender=sender, **named)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/core/signals/handlers.py", line 7, in on_order_created
    print(**kwargs['order'])
TypeError: print() argument after ** must be a mapping, not Order
2024-05-08 20:27:04,591 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:12,731 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:26,150 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:37,139 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-08 20:27:48,865 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:50,929 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:53,013 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:55,086 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:57,173 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:27:59,260 (WARNING) - django.request - Not Found: /api/carts/31e197b4-dd56-402a-bd35-8e4f984a7d92/
2024-05-08 20:28:39,084 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 20:28:39,086 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 20:28:45,321 (WARNING) - django.request - Unauthorized: /api/orders/
2024-05-08 20:28:45,325 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 20:28:45,326 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 20:29:25,604 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 21:33:37,304 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:33:37,306 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:34:31,410 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:34:38,939 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:34:45,115 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:34:56,322 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:35:02,137 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:35:14,464 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:37:28,779 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:37:28,783 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:40:36,061 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:40:36,063 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:40:39,500 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:40:39,510 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:03,083 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:03,088 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:19,218 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:25,184 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:25,186 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:37,085 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:41:37,088 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:42:03,042 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:42:20,749 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 21:42:25,443 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 21:42:46,602 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 21:43:26,019 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:43:27,887 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:44:18,378 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 21:44:20,269 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:18,852 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:18,854 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:24,446 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:24,448 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:28,320 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:28,322 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:52,018 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:03:52,020 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:05:26,710 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:05:43,052 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:10,144 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:12,126 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:12,134 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:15,127 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:15,128 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:30,114 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:30,116 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:07:54,710 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:10:48,644 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:04,651 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:10,161 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:12,139 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:12,141 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:23,435 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:23,438 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:47,406 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:11:58,556 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:37,991 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:37,995 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:37,997 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:37,999 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:41,762 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:41,765 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:42,319 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:42,321 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:47,519 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:47,523 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:47,707 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:47,708 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:48,081 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:12:48,083 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:13:54,651 (WARNING) - django.request - Not Found: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/
2024-05-08 22:13:59,882 (WARNING) - django.request - Not Found: /api/carts/fb61ced7-9c56-4e75-93e6-325f6b244833/
2024-05-08 22:14:01,088 (WARNING) - django.request - Method Not Allowed: /api/carts/
2024-05-08 22:14:14,671 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-08 22:14:16,557 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-09 13:52:49,420 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-09 13:53:03,410 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-09 13:53:05,738 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-09 13:53:08,071 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-09 13:53:10,398 (WARNING) - django.request - Not Found: /api/carts/null/
2024-05-09 14:51:50,560 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-09 14:52:19,180 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
2024-05-09 14:52:23,533 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:23,551 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:27,453 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:27,476 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:39,054 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:39,084 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:48,430 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:52:48,448 (WARNING) - django.request - Not Found: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/
2024-05-09 14:53:01,904 (ERROR) - django.request - Internal Server Error: /api/carts/e1213de2-1694-4b70-877b-9a3dd9caea5d/items/
Traceback (most recent call last):
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py", line 90, in save
    cart_item = CartItem.objects.get(
                ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 649, in get
    raise self.model.DoesNotExist(
api.models.CartItem.DoesNotExist: CartItem matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
psycopg2.errors.ForeignKeyViolation: insert or update on table "api_cartitem" violates foreign key constraint "api_cartitem_cart_id_26c2013b_fk_api_cart_id"
DETAIL:  Key (cart_id)=(e1213de2-1694-4b70-877b-9a3dd9caea5d) is not present in table "api_cart".


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/core/handlers/base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/views/decorators/csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/viewsets.py", line 124, in view
    return self.dispatch(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 509, in dispatch
    response = self.handle_exception(exc)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 469, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 480, in raise_uncaught_exception
    raise exc
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/views.py", line 506, in dispatch
    response = handler(request, *args, **kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 19, in create
    self.perform_create(serializer)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/rest_framework/mixins.py", line 24, in perform_create
    serializer.save()
  File "/Users/sjo/Documents/Projects/Nanocellum/backend/api/serializers.py", line 99, in save
    self.instance = CartItem.objects.create(
                    ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 679, in create
    obj.save(force_insert=True, using=self.db)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 822, in save
    self.save_base(
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 909, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 1067, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/base.py", line 1108, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/query.py", line 1847, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/models/sql/compiler.py", line 1823, in execute_sql
    cursor.execute(sql, params)
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/Users/sjo/miniforge3/envs/BE/lib/python3.11/site-packages/django/db/backends/utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
django.db.utils.IntegrityError: insert or update on table "api_cartitem" violates foreign key constraint "api_cartitem_cart_id_26c2013b_fk_api_cart_id"
DETAIL:  Key (cart_id)=(e1213de2-1694-4b70-877b-9a3dd9caea5d) is not present in table "api_cart".

2024-05-09 14:53:27,997 (INFO) - django.utils.autoreload - /Users/sjo/Documents/Projects/Nanocellum/backend/backend/settings.py changed, reloading.
2024-05-09 14:54:11,472 (INFO) - django.utils.autoreload - Watching for file changes with StatReloader
